src/transport.c

```diff
@@ -108,10 +108,13 @@ static int find_address(unsigned char ipaddr[BOOTH_IPADDR_LEN],
 				break;
 
 		if (matched == node->addrlen) {
-			*address_bits_matched = matched * 8;
-			*me = node;
-			did_match = EXACT_MATCH;
-			break;
+			if((matched_tmp < matched * 8)||((node->type == SITE)&&(matched_tmp == matched * 8)))
```

Comment:
It sounds like it may pick a wrong one as the "local site" if the sites and the arbitrator are in the same subnet. There seems to be some problem in the logic around the "EXACT_MATCH".